components/blitz/src/omero/gateway/Gateway.java

```diff
@@ -1220,7 +1234,9 @@ private void keepSessionAlive() throws DSOutOfServiceException {
             c = i.next();
             if (c.needsKeepAlive()) {
                 if (!c.keepSessionAlive()) {
-                    throw new DSOutOfServiceException("Network not available");
+                    // Session has died, e. g. due to server restart.
```

Comment:
Should work as before (show connection warning and reconnect again, when possible), tested on the original PR #5513 (point 3). 

As far as I remember the Exception was the problem, because this reconnect mechanism only worked for a stale network connection, but not when the server was restarted. Then the Exception crashed Insight. With this change the stale connector is quietly removed and a new connector will be created.